File: 3_Software_Cloud/iov-full-stack-app/backend/source/src/main/java/de/rub/enesys/ovms3lite/model/Vehicle.java
package de.rub.enesys.ovms3lite.model;

import lombok.*;

import javax.persistence.*;
import java.util.List;


@Entity
@ToString
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Vehicle {


    @Id
    @Column(length = 50, unique = true)
    private String id;

    @Column(length = 50, unique = true)
    private String licencePlate;

//    @Column(length = 50, unique = true)
    private String iconColor;

    private String model;
    private String make;

    @OneToMany(mappedBy = "vehicle",cascade = CascadeType.ALL,orphanRemoval = true)
    List<VehicleEvent> vehicleEventList;

    @OneToMany(mappedBy = "vehicle",cascade = CascadeType.ALL,orphanRemoval = true)
    List<Location> locationList;

}

File: 3_Software_Cloud/iov-full-stack-app/frontend-map/iov-frontend/src/app/model/Vehicle.ts
export class Vehicle {
    make: string;
    model: string;
    id: string;
    licencePlate: string;
    iconColor: string;

    // vehicleStatus: VehicleStatus;
    carOn: boolean;
    soc: number;
    odometer: number;
    chargingState: number;

    eventList: Event[];


    locationList: Location[];
}

export class VehicleStatus {
    carOn: boolean;
    soc: number;
    odometer: number;
    chargingState: number;
}